src/jit/codegenlegacy.cpp

```diff
@@ -18986,7 +18994,11 @@ regMaskTP CodeGen::genCodeForCall(GenTreeCall* call, bool valUsed)
                     }
 
                     regNumber tcbReg;
-                    tcbReg = genPInvokeCallProlog(frameListRoot, nArgSize, methHnd, returnLabel);
+
+                    if (!compiler->opts.ShouldUsePInvokeHelpers())
+                    {
+                        tcbReg = genPInvokeCallProlog(frameListRoot, nArgSize, methHnd, returnLabel);
```

Comment:
tcbReg is going to contain garbage after this, since it won't be set. However, it is used below.